frame/merkle-mountain-range/src/mmr/storage.rs

```diff
@@ -17,36 +17,73 @@
 
 //! A MMR storage implementations.
 
-use codec::Encode;
-use frame_support::traits::Get;
+use codec::{Decode, Encode, MaxEncodedLen};
 use mmr_lib::helper;
-use sp_core::offchain::StorageKind;
-use sp_io::{offchain, offchain_index};
-use sp_runtime::traits::UniqueSaturatedInto;
+use sp_core::{bounded::BoundedVec, offchain::StorageKind, Get};
+use sp_io::offchain;
 use sp_std::iter::Peekable;
 #[cfg(not(feature = "std"))]
 use sp_std::prelude::*;
 
 use crate::{
 	mmr::{utils::NodesUtils, Node, NodeOf},
 	primitives::{self, NodeIndex},
-	Config, Nodes, NumberOfLeaves, Pallet,
+	Config, NewNodes, NumberOfLeaves, Pallet, Peaks,
 };
 
+/// Leaf (data) and parent nodes (hashes) added by current block.
+///
+/// This is single value storage overwritten by each block. It is used as temporary
+/// runtime storage until after block is built and offchain worker can move/copy this
+/// leaf in offchain database under `node_offchain_key()`.
+#[derive(Encode, Decode, MaxEncodedLen, scale_info::TypeInfo)]
+#[scale_info(skip_type_params(MaxLeafSize, MaxMmrHeight))]
+pub struct NewLeafNodesBatched<Hash, MaxLeafSize, MaxMmrHeight> {
```

Comment:
yeah, need to find better name.. I would avoid `pending` because it's not actually pending from the runtime's point of view - from runtime's point of view that storage item actually stores `NewLeafAndNodes` - maybe that'd be a better name?

Reply:
I was referring to `pending` because it's in temporary storage and pending further processing, but you're right, for the runtime it's actually a new leaf. The part with `LeafAndNodes` sounds complicated to me. Maybe `NewLeafSubtree` ? Although this can be misinterpreted. It's basically a part of the tree that's not in the offchain storage yet. I don't know. Can't think of better names now, but I don't have strong preferences here.